router/worker.go

```diff
@@ -799,9 +799,6 @@ func (w *worker) proxyRequest(ctx context.Context, destinationJob types.Destinat
 
 		proxyRequestResponse.RespStatusCodes, proxyRequestResponse.RespBodys = w.prepareResponsesForJobs(&destinationJob, respStatusCode, respBodyTemp)
 		proxyRequestResponse.RespContentType = contentType
-	} else if oauthV2Enabled {
-		proxyRequestResponse.RespStatusCodes, proxyRequestResponse.RespBodys = w.prepareResponsesForJobs(&destinationJob, proxyRequestResponse.ProxyRequestStatusCode, proxyRequestResponse.ProxyRequestResponseBody)
-		proxyRequestResponse.RespContentType = http.DetectContentType([]byte(proxyRequestResponse.ProxyRequestResponseBody))
```

Comment:
Didn't understood why we are removing this?

Reply:
What if `transportResponse.InterceptorResponse.StatusCode` is 0 for a successful response [here](https://github.com/rudderlabs/rudder-server/pull/4653/files#diff-69dca86868b1232b1aae3107d5590adf53da2a1bc93c414874ad54a815c59e56R443-R449)?

Reply:
>Didn’t understood why we are removing this?
The response contract for `/v1/<DEST_TYPE>/proxy` endpoint in transformer is
```json
200 OK
{
  “output”: {
    “message”: “”,
     “response”: [
           {“statusCode”: 400, “metadata”: {jobId: 1}},
           {“statusCode”: 502, “metadata”: {jobId: 2}},
      ],
     “authErrorCategory”: “”
  }
}
```
When there is no problem with LB, we would always get `200` from the endpoint. And based on the logic of `prepareResponsesForJobs`, we would set proxy response status-code(200) to all the jobs.
Before this change, jobId: 1 would be set with `200` instead of `400` & jobId: 2 would be set with `200` instead of `500`

This is the reason we are removing this part of source-code. Let me know if it makes sense